deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/optimize/listeners/CollectScoresIterationListener.java

```diff
@@ -37,7 +37,74 @@
 
     private int frequency;
     private int iterationCount = 0;
-    private List<Pair<Integer, Double>> scoreVsIter = new ArrayList<>();
+    //private List<Pair<Integer, Double>> scoreVsIter = new ArrayList<>();
+
+    public static class ScoreStat {
+        public static final int BUCKET_LENGTH = 10000;
+
+        private int position = 0;
+        private int bucketNumber = 1;
+        private List<long[]> indexes;
+        private List<double[]> scores;
+
+        public ScoreStat() {
+            indexes = new ArrayList<>(1);
+            indexes.add(new long[BUCKET_LENGTH]);
+            scores = new ArrayList<>(1);
+            scores.add(new double[BUCKET_LENGTH]);
+        }
+
+        public List<long[]> getIndexes() {
+            return indexes;
+        }
+
+        public List<double[]> getScores() {
+            return scores;
+        }
```

Comment:
For better usability, let's provide a method like indexes() and scores() that returns long[] and double[] with only the real data, without the extra preallocated bit at the end.
With the current methods, in almost all cases in practice, users will get a list of length 1, with an array of unknown "real" length - i.e., long[]/double[] with a bunch of 0s at the end.

On the off chance that users have more than Integer.MAX_VALUE total entries (should almost never happen in practice) we can just say "use getIndexes() and getPosition()"

Reply:
Methods and test added.